src/lib_json/json_value.cpp

```diff
@@ -1172,43 +1172,43 @@ Value const& Value::operator[](CppTL::ConstString const& key) const {
 Value& Value::append(const Value& value) { return (*this)[size()] = value; }
 /// \brief Insert value in array at specific index
 bool Value::insert(ArrayIndex index, const Value& newValue) {
-  if(type() != arrayValue) {
+  if (type() != arrayValue) {
     return false;
   }
-  if(!isValidIndex(index)) {
+  if (!isValidIndex(index)) {
     return false;
   }
   ArrayIndex oldsize = size();
-  resize(oldsize+1);
+  resize(oldsize + 1);
   ArrayIndex length = size();
-  if(length != oldsize +1) {
-    return false;
-  }
-  for(ArrayIndex i = length ; i> index ; i--) {
-    (*this)[i] = (*this)[i-1];
+
+  for (ArrayIndex i = length; i > index; i--) {
+    CZString key(i);
+    (*value_.map_)[key] = (*this)[i - 1];
   }
-  (*this)[index]= newValue;
+  (*this)[index] = newValue;
   return true;
 }
 #if JSON_HAS_RVALUE_REFERENCES
-Value& Value::append(Value&& value) { return (*this)[size()] = std::move(value); }
-bool Value::insert(ArrayIndex index, const Value&& newValue) {
-  if(type() != arrayValue) {
+Value& Value::append(Value&& value) {
+  return (*this)[size()] = std::move(value);
+}
+bool Value::insert(ArrayIndex index, Value&& newValue) {
```

Comment:
we should reuse the logic of the const Value& overload instead of duplicating it.